File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        lo, hi = 0, len(numbers) - 1

        while lo < hi:
            temp = numbers[lo] + numbers[hi]
            if temp < target: lo += 1
            elif temp > target: hi -= 1
            else: return [lo + 1, hi + 1]

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int lo = 0;
        int hi = numbers.length - 1;

        while (lo < hi) {
            int chk = numbers[lo] + numbers[hi];
            if (chk == target) {
                break;
            } else if (chk < target) {
                lo++;
            } else {
                hi--;
            }
        }

        int[] result = {lo + 1, hi + 1};
        return result;
    }
}